apis/v1alpha2/gateway_types.go

```diff
@@ -26,6 +26,8 @@ import (
 // +kubebuilder:subresource:status
 // +kubebuilder:storageversion
 // +kubebuilder:printcolumn:name="Class",type=string,JSONPath=`.spec.gatewayClassName`
+// +kubebuilder:printcolumn:name="Address",type=string,JSONPath=`.status.addresses[*].value`
```

Comment:
Chatted a bit about this, I'm a bit concerned that this would lead to more confusion than it's worth since it will only be able to print the first address. I'd lean towards removing this instead.

Reply:
Actually maybe this is fine. It will be remarkably better for >99% of use cases where there will only be one address, so optimizing for that probably makes sense.

Reply:
Is there a way to list this only when `-o wide` is applied?

> Actually maybe this is fine. It will be remarkably better for >99% of use cases where there will only be one address, so optimizing for that probably makes sense.

In that case, why not do `addresses[0].value`?

Reply:
I think it's fine to include this in regular output to match ingress, agree that `addresses[0].value` makes more sense given that that's how `*` would effectively be interpreted.